editor.planx.uk/src/pages/Team/components/AddFlow/types.ts

```diff
@@ -0,0 +1,41 @@
+import { object, string } from "yup";
+
+// TODO: Validate unique flow name
+export const validationSchema = object().shape({
+  mode: string().oneOf(["new", "copy", "template"]).required(),
+  flow: object({
+    slug: string().required("Slug is required"),
+    name: string().required("Name is required"),
+  }).required(),
+  id: string().when("mode", {
+    is: "new",
+    then: string().notRequired(),
+    otherwise: string().required("ID is required for copy and template modes"),
+  }),
+});
+
+type NewFlow = {
+  slug: string;
+  name: string;
+  sourceId?: string;
+};
+
+export type CreateFlow = {
+  mode: "new" | "copy" | "template";
+  flow: NewFlow;
+};
+
+export const CREATE_FLOW_MODES = [
+  {
+    mode: "new",
+    title: "New flow",
+  },
+  {
+    mode: "copy",
+    title: "From copy...",
+  },
+  {
+    mode: "template",
+    title: "From template...",
+  },
+] as const;
```

Comment:
I think there's a lot of room for improvement here with regard to the copy used. I think we'll need a one or two line explainer text somewhere?